torch/csrc/autograd/functions/batch_normalization.cpp

```diff
@@ -123,6 +124,7 @@ auto BatchNormBackward::apply(const variable_list& grad_outputs) -> variable_lis
   use_cudnn = (input.type().backend() == at::kCUDA
                && input.type().scalarType() != at::kHalf
                && weight.defined() && bias.defined() && training
+               && input.size(0) <= 131070
```

Comment:
We should add some comments here. I would have no idea what is even going on if I didn't notice the issue you posted before, and I'm pretty sure no one will remember than in a few weeks.